src/test/java/edu/cornell/kfs/concur/batch/service/impl/ConcurExpenseV3ServiceUpdateReportStatusTest.java

```diff
@@ -0,0 +1,362 @@
+import edu.cornell.kfs.sys.web.mock.MockMvcWebServerExtension;
+
+@Execution(ExecutionMode.SAME_THREAD)
+public class ConcurExpenseV3ServiceUpdateReportStatusTest {
+
+    private static final String TEST_BEARER_TOKEN = "ABCDEFG1234567abcdefg1234567ABCDEFG1234567";
+    private static final String INVALID_BEARER_TOKEN = "0BCDEFG1234567abcdefg1234567ABCDEFG1234567";
+    private static final String NON_EXISTING_REPORT_ID_1 = "aEiOuYYYYYYYYYY";
+    private static final String NON_EXISTING_REPORT_ID_2 = "bcdFGHjklmnPQRS";
+    private static final String NON_EXISTING_REPORT_ID_3 = "777777777777777";
+    private static final String TRAVELER_NAME_JOHN_DOE = "John Doe";
+    private static final String TRAVELER_EMAIL_JOHN_DOE = "johndoe123@somedomain.com";
+    private static final String MESSAGE_ACTION_TAKEN = "Action has been taken";
+    private static final String MESSAGE_INACTIVE_CHART = "Chart code is inactive";
+    private static final String MESSAGE_MISSING_ACCOUNT = "Account number is missing";
+    private static final String MESSAGE_ERROR_ENCOUNTERED = "Encountered an error while validating";
+    private static final String INVALID_WORKFLOW_ENDPOINT = "/expensereports/v4/unknown";
+
+    @RegisterExtension
+    static MockMvcWebServerExtension webServerExtension = new MockMvcWebServerExtension();
+
+    private static MockConcurExpenseV4WorkflowController mockEndpoint;
+
+    private String accessToken;
+    private ConcurBatchUtilityService mockConcurBatchUtilityService;
+    private TestConcurExpenseV3ServiceImpl concurExpenseV3Service;
+
+    @BeforeAll
+    static void setUpMockEndpoint() throws Exception {
+        mockEndpoint = new MockConcurExpenseV4WorkflowController(TEST_BEARER_TOKEN);
+        webServerExtension.initializeStandaloneMockMvcWithControllers(mockEndpoint);
+    }
+
+    @BeforeEach
+    void setUp() throws Exception {
+        this.accessToken = TEST_BEARER_TOKEN;
+        this.mockConcurBatchUtilityService = createMockConcurBatchUtilityService(webServerExtension.getServerUrl());
+        
+        this.concurExpenseV3Service = new TestConcurExpenseV3ServiceImpl();
+        concurExpenseV3Service.setConcurBatchUtilityService(mockConcurBatchUtilityService);
+        concurExpenseV3Service.setConcurEventNotificationV2WebserviceService(
+                createConcurEventNotificationV2WebserviceService(mockConcurBatchUtilityService));
+        concurExpenseV3Service.setConfigurationService(createMockConfigurationService());
+        concurExpenseV3Service.setConcurAccountValidationService(Mockito.mock(ConcurAccountValidationService.class));
+        concurExpenseV3Service.setSimulateProduction(true);
+        
+        mockEndpoint.addReportsAwaitingAction(ConcurTestConstants.REPORT_ID_1,
+                ConcurTestConstants.REPORT_ID_2, ConcurTestConstants.REPORT_ID_3);
+    }
+
+    @AfterEach
+    void tearDown() throws Exception {
+        concurExpenseV3Service = null;
```

Comment:
Is the "throws Exception" needed for all or any of the methods within this test class?

Reply:
Many of our other unit tests already have a convention of adding `throws Exception` to methods annotated with `@Before`/`@BeforeEach` or `@After`/`@AfterEach`. I would prefer to keep that part of the declaration in place.